modules/ml/src/main/java/org/apache/ignite/ml/sql/SQLFunctions.java

```diff
@@ -39,10 +70,15 @@
      */
     @QuerySqlFunction
     public static double predict(String mdl, Double... x) {
-        System.out.println("Prediction for " + Arrays.toString(x));
+        Model<Vector, Double> infMdl;
 
-        try (Model<Vector, Double> infMdl = IgniteModelStorageUtil.getModel(mdl)) {
-            return infMdl.predict(VectorUtils.of(x));
+        synchronized (cache) {
```

Comment:
may be we can use double-check?
synchronization on every predict can be expensive I think